set-web/src/main/java/eu/europeana/set/web/service/impl/BaseUserSetServiceImpl.java

```diff
@@ -488,6 +492,14 @@ void validateControlledValues(UserSet webUserSet) throws RequestBodyValidationEx
 	    throw new RequestBodyValidationException(UserSetI18nConstants.USERSET_VALIDATION_PROPERTY_VALUE,
 		    new String[] { WebUserSetModelFields.TYPE, webUserSet.getType() });
 	}
+	// if type is BookmarkFolder or Collection, validate subject
+	// subject must not contain entity reference
+	if(webUserSet.getType() != null && ! isEntityBestItemsSet(webUserSet)) {
+		if(webUserSet.isEntityBestItemSet()) {
+			throw new RequestBodyValidationException(UserSetI18nConstants.USERSET_VALIDATION_PROPERTY_VALUE,
```

Comment:
there is a different message which is specific for missing mandatory fields. Please switch to the other message by using appropriate parameters

Reply:
Isn't this validation part of the validateBestItemsSet?

Reply:
this validation is for BookmarkFolder and Collection type user set. As they cannot have a entity reference in subject field

Reply:
Have changed the message